### PR TITLE
Makefile: Add kind-image target

### PR DESCRIPTION
**Pull request by joestringer**
Add a new kind-image target, mirroring the microk8s target for building
and tagging the image.

To use:

    $ make kind
    <install Cilium>
    $ make kind-image
    $ kubectl -n kube-system set image daemonset/cilium cilium-agent=localhost:5000/cilium/cilium-dev:local

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -488,16 +488,25 @@ check-microk8s: ## Validate if microk8s is ready to install cilium.
...
 
 LOCAL_IMAGE_TAG=local
-LOCAL_IMAGE=localhost:32000/$(DOCKER_DEV_ACCOUNT)/cilium-dev:$(LOCAL_IMAGE_TAG)
-microk8s: check-microk8s ## Build cilium-dev docker image and import to mircrok8s
+microk8s: export DOCKER_REGISTRY=localhost:32000
+microk8s: export LOCAL_IMAGE=$(DOCKER_REGISTRY)/$(DOCKER_DEV_ACCOUNT)/cilium-dev:$(LOCAL_IMAGE_TAG)
+microk8s: check-microk8s ## Build cilium-dev docker image and import to microk8s
 	$(QUIET)$(MAKE) dev-docker-image DOCKER_IMAGE_TAG=$(LOCAL_IMAGE_TAG)
 	@echo "  DEPLOY image to microk8s ($(LOCAL_IMAGE))"
-	$(QUIET)$(CONTAINER_ENGINE) tag $(IMAGE_REPOSITORY)/cilium-dev:$(LOCAL_IMAGE_TAG) $(LOCAL_IMAGE)
```
Comment (glibsm) [inline Makefile:495]: this change doesn't seem related to the commit message

**Review by glibsm — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (joestringer) [reply Makefile:495]: Yeah arguably it should be two commits, rework the microk8s target and add the kind target. When we move the LOCAL_IMAGE declaration out of the top-level of the Makefile into the target-specific variable, it means we no longer need this step.

Comment (joestringer) [reply Makefile:495]: ^^ Split out for better clarity.

**Comment by joestringer** [comment]
Marking for backport to make v1.11 debugging easier in future.
